clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -68,14 +107,16 @@ public MembershipManagerImpl(String groupId, String groupInstanceId, AssignorSel
         }
         this.groupInstanceId = Optional.ofNullable(groupInstanceId);
         this.targetAssignment = Optional.empty();
-        this.nextTargetAssignment = Optional.empty();
+        this.log = logContext.logger(MembershipManagerImpl.class);
     }
 
     /**
      * Update assignor selection for the member.
      *
-     * @param assignorSelection New assignor selection
-     * @throws IllegalArgumentException If the provided assignor selection is null
+     * @param assignorSelection New assignor selection. If empty is provided, this will
+     *                          effectively clear the previous assignor selection defined for the
+     *                          member.
+     * @throws IllegalArgumentException If the provided optional assignor selection is null.
      */
     public final void setAssignorSelection(AssignorSelection assignorSelection) {
```

Comment:
btw, we can't change the assignor while the consumer runs. We could probably remove this method or make it private.

Reply:
Good point, this is only defined when the consumer is created. Removed.